### PR TITLE
Pass the knex connection to the repositories

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
This is the correct way to do this rather then create a new connection pool per repository

### How to review 
Have a look at the implementation, Ive only made this change for 1 Repository so far. But open to feedback

### CODE REVIEW

**Comment by tgandrews** [comment]
Doesn't look too bad but this is going to have a big impact on existing PRs so we need to manage how we do it.

**Comment by tgandrews** [comment]
@jonnyshaw89 do you have prettier installed? There are a lot of formatting changes in this.

File: eq-author-api/config/knexfile.js [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
   },
 
   test: {
     client: "postgresql",
-    connection: process.env.DB_CONNECTION_URI
+    connection: process.env.DB_CONNECTION_URI,
+    pool: {
+      min: 2,
+      max: 10
+    }
```
Comment (samiwel) [inline eq-author-api/config/knexfile.js]: Can we do away with separate configurations or is one specifically required for test?
If they're the same maybe they can be combined into one config:

```js
const baseDbConfig = {
	client: "postgresql",
    connection: process.env.DB_CONNECTION_URI,
    pool: {
      min: 2,
      max: 10
    }
}

module.exports = {
	development: baseDbConfig,
    test: baseDbConfig,
    production: {
	  ...baseDbConfig,
      pool: {
        min: 2,
        max: 20
      }
   }
}

```

Comment (jonnyshaw89) [reply eq-author-api/config/knexfile.js]: 👍 already done

**Comment by jonnyshaw89** [comment]
@samiwel @tgandrews Can you guys have another look over this one please?

**Comment by adammahmood** [comment]
Is this really necessary?

https://stackoverflow.com/questions/49588672/knex-js-sql-knex-sql-connection-pools

**Pull request merged**